File: app/src/main/java/fhu/bughousechess/pieces/Empty.java
package fhu.bughousechess.pieces;

import android.widget.ImageView;

import java.util.Set;

import fhu.bughousechess.Move;
import fhu.bughousechess.pieces.Piece;

public class Empty extends Piece
{
    public Empty()
    {
        color = "";
        type = "";
        wasPawn = false;
        empty = true;
    }

    @Override
    public int getResID()
    {
        return android.R.color.transparent;
    }

    @Override
    public Set<Move> getMoves(Piece[][] positions, int x, int y, int boardNumber)
    {
        return null;
    }
}

File: app/src/main/java/fhu/bughousechess/pieces/Empty.kt
package fhu.bughousechess.pieces

import android.R
import fhu.bughousechess.Move

class Empty : Piece() {
    init {
        color = ""
        type = ""
        wasPawn = false
        empty = true
    }

    override val resID: Int
        get() = R.color.transparent

    override fun getMoves(
        positions: Array<Array<Piece?>?>?,
        x: Int,
        y: Int,
        boardNumber: Int
    ): Set<Move?>? {
        return null
    }
}